### ISSUE TITLE
bug(merklesync-updater): tries to poll new_root before it has finished syncing all messages

### ISSUE THREAD

**Issue by ltchang2019** [nomad-xyz/rust#108]
TL;DR the bug in the updater that failed all dev homes is as follows:
- Updater tries produce  new_roots from a local merkle tree which is built from all the emitted messages for that home (same process as the processor)
- The updater has until now never synced/indexed messages
- The updater started syncing messages but before that has finished, it picks the current root of the local tree and uses that as its new_root
- This update of last_committed-->new_root is invalid (the new_root is actually a really old root)
- The improper update is submitted and the home is failed (for each updater)

Solution:
- Have the syncing merkle tree keep track of the current `committed_root` corresponding to its current root
- I.e. Each time the syncing tree stores a new leaf, it will get the` Dispatch.committed_root` field and update its current `committed_root` field
- The updater will only use `tree.root()` if `tree.current_committed() == old_root`
